tinygrad/lazy.py

```diff
@@ -52,6 +52,10 @@ def realized(self) -> Optional[Buffer]:
   @property
   def base(self) -> LazyBuffer: return self._base if self._base is not None else self
 
+  @functools.cached_property
+  def key(self) -> bytes: return hashlib.sha256(functools.reduce(lambda x,y: x+y, [s.base.key for s in self.srcs if s.base.realized is None],
```

Comment:
Yea, this doesn't work, crypto hash functions are super slow. We have to find the source of the nondeterminism and tiebreak it there.